Modernization - May 2024

dbaty opened the pull request:
Commits should be reviewed one by one.

dbaty committed:
Remove support of Python 3.7

Active support of Python 3.7 has ended in June 2020. Security support
has ended in June 2023.

dbaty committed:
req: Remove now useless extra dep of isort from dev dependencies

The "pyproject" extra dependency has been removed from isort 5.0.0,
which has been released in July 2020. It caused a warning with when a
more recent version was installed:

    $ pip install -r requirements_dev.txt
    [...]
    WARNING: isort 5.13.2 does not provide the extra 'pyproject'
    [...]

dbaty committed:
Add support of Python 3.12

dbaty committed:
Remove support of Git 2.18 and earlier versions

We maintained non-obvious code to support versions of Git that did not
have the `--only-matching` option, which appeared in Git 2.19. This
version was released in September 2018, it's time to move on.

dbaty committed:
qa: Move pylint configuration to `pyproject.toml`

That's where all tools are configured (except ReadTheDocs, which does
not support `pyproject.toml`).

dbaty committed:
qa: Enable more pylint checks

These checks seem useful, I don't know why they should disabled.

dbaty committed:
tests: Add unit tests for `annotations.get_known_future_tags()`

dbaty committed:
req: Remove optional dependency on `toml`

It was not necessary with Python >= 3.11, which ships with the
`tomllib` package in the standard library.

Also, the extra requirement could be a bit confusing, especially for
non-Python developers. "Do I need it now, will I need it later?"

For the sake of simplicity, the extra requirement has been removed.
We now always install `toml` on Python versions earlier than 3.11 (and
never on 3.11 and above).

sandre35 commented on .pylintrc:
```diff
@@ -1,76 +0,0 @@
-
-# Template used to display messages. This is a python new-style format string
-# used to format the message information. See doc for all details
-msg-template="{path}:{line}: [{msg_id}({symbol}), {obj}] {msg}"
```
There is no copy of the REPORTS section. Is this done on purpose?

dbaty replied:
Yes, this is intentional. I am switching to the default template for simplicity's sake. I find it readable enough: `{path}:{line}:{column}: {msg_id}: {msg} ({symbol})`. Example:

    src/check_oldies/commands.py:33:4: W0612: Unused variable 'a' (unused-variable)

TiphaineLAURENT commented on tests/data/project7/file1.py:
```diff
@@ -0,0 +1,2 @@
+# TIMEBOMB: FEWTURE-BOOM1
+# FEWTURE-DO-NOT-REPORT: do not report, it's not within an annotation
```
J'ai bien envie de d'utiliser le terme `FEWTURE` premier degré maintenant 😄

TiphaineLAURENT commented on tests/test_annotations.py:
```diff
@@ -65,3 +67,26 @@ def test_get_login_from_committer_email():
+            future_tag_regex=base.TESTING_FUTURE_TAG,
+            whitelist=(),
+        )
+        assert tags == {"FEWTURE-BOOM1", "FEWTURE-BOOM2"}
```
Est-ce que le tag n'est pas `TIMEBOMB` ici ?

dbaty replied:
Le tag cherché est bien de type "FUTURE-xxx".

"TIMEBOMB", c'est en fait un "FIXME". Peut-être que je devrais le nommer "FIIXME" dans les tests, ce sera plus proche de "FIXME" et donc plus clair.

dbaty committed:
tests: Add minimal tests for `githost.GitHubApi`

dbaty committed:
qa: Introduce mypy

dbaty committed:
docs: Replace use of old `pkg_resources` by `importlib`

`pkg_resources` is no more.

dbaty committed:
Remove support of Python 3.8 and 3.9

They both reached their end of life.

dbaty committed:
Add support of Python 3.13 and 3.14

dbaty committed:
qa: Disable pylint's `too-many-positional-arguments` warning

Just like other "too-many-*" warnings, I don't like them. I suppose
that we'll "just know" when a function has too many arguments.

dbaty committed:
qa: Fix spurious "pylint: disable" comment

It's not necessary.

dbaty committed:
qa: Run `pyupgrade --py310-plus`

dbaty committed:
build: Move contents of `setup.cfg` to `pyproject.toml`

dbaty committed:
docs: Update authorship and maintainership

dbaty committed:
build: Be bold and declare the project as stable

It has barely moved in years: is that not what stability means? ;)

dbaty committed:
qa: Don't report about "FIXME" in `CHANGES.rst`

It's unlikely to have such annotations. However, it does contain the
word "FIXME", which we want to ignore.

dbaty committed:
build: Pin `toml` optional dependency
